src/frontends/tensorflow_common/src/op/tensor_list_operations.cpp

```diff
@@ -132,29 +146,82 @@ OutputVector translate_tensor_list_push_back_op(const NodeContext& node) {
+    tensor = make_shared<v0::Unsqueeze>(tensor, zero_const);
 
     // insert the tensor into the end
-    auto updated_list = make_shared<Concat>(OutputVector{input_handle, tensor}, 0);
+    auto updated_list = make_shared<v0::Concat>(OutputVector{input_handle, tensor}, 0);
 
     set_node_name(node.get_name(), updated_list);
     return {updated_list};
 }
 
+OutputVector translate_tensor_list_resize_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"TensorListResize"});
+    auto input_handle = node.get_input(0);
+    auto size = node.get_input(1);
+
+    // create auxiliary constants
+    auto zero_const = make_shared<v0::Constant>(element::i32, Shape{1}, 0);
+    auto one_const = make_shared<v0::Constant>(element::i32, Shape{1}, 1);
+    auto max_const = make_shared<v0::Constant>(element::i32, Shape{1}, numeric_limits<int32_t>::max());
+
+    // compute the current length of the list and item shape
+    auto tensor_list_shape = make_shared<v3::ShapeOf>(input_handle, element::i32);
+    auto list_length = make_shared<v8::Slice>(tensor_list_shape, zero_const, one_const, one_const);
+    auto item_shape = make_shared<v8::Slice>(tensor_list_shape, one_const, max_const, one_const);
+
+    // compute a size of the dummy tensor to resize
+    // and clip it by zero if it is negative
+    Output<Node> dummy_tensor_size = make_shared<v1::Subtract>(size, list_length);
+    dummy_tensor_size = make_shared<v1::Maximum>(dummy_tensor_size, zero_const);
+
+    // create dummy tensor and concatenate it
+    auto zero_const_same_type = create_same_type_const<float>(input_handle, vector<float>{0.0f}, Shape{});
+    auto dummy_tensor_shape = make_shared<v0::Concat>(OutputVector{dummy_tensor_size, item_shape}, 0);
+    auto dummy_tensor = make_shared<v3::Broadcast>(zero_const_same_type, dummy_tensor_shape);
+    input_handle = make_shared<v0::Concat>(OutputVector{input_handle, dummy_tensor}, 0);
+
+    // reshape size to have 1D tensor with one element
+    auto new_size_shape = make_shared<v0::Constant>(element::i32, Shape{1}, 1);
+    size = make_shared<v1::Reshape>(size, new_size_shape, false);
+
+    // resize can also shrink the input tensor list
+    input_handle = make_shared<v8::Slice>(input_handle, zero_const, size, one_const);
+
+    set_node_name(node.get_name(), input_handle.get_node_shared_ptr());
+    return {input_handle};
+}
+
+OutputVector translate_tensor_list_length_op(const NodeContext& node) {
+    default_op_checks(node, 1, {"TensorListLength"});
+    auto input_handle = node.get_input(0);
+
+    // create auxiliary constants
+    auto zero_const = make_shared<v0::Constant>(element::i32, Shape{1}, 0);
+    auto one_const = make_shared<v0::Constant>(element::i32, Shape{1}, 1);
+
+    // compute the current length of the list
+    auto tensor_list_shape = make_shared<v3::ShapeOf>(input_handle, element::i32);
+    auto list_length = make_shared<v8::Slice>(tensor_list_shape, zero_const, one_const, one_const);
```

Comment:
What will be result for empty list? In pytorch frontend for similar op we had to use `ReduceSum` to return 0 for empty list https://github.com/openvinotoolkit/openvino/blob/498731f8fd773fbd8e35f953522f7721f726bfc9/src/frontends/pytorch/src/op/len.cpp#L28

Reply:
Empty list means to have the first dimension to be zero `0`. So the slice of this shape will be zero.

Reply:
Shape will be not zero, it will be [], so slice will return empty tensor. Could you add a test for empty list?

Reply:
Test was added